deepset_cloud_sdk/workflows/pipeline_client/pipeline_service.py

```diff
@@ -0,0 +1,239 @@
+"""Pipeline publishing service for deepset Cloud SDK."""
+# pylint: disable=unnecessary-ellipsis,import-outside-toplevel
+import asyncio
+from io import StringIO
+from typing import Any, Protocol, runtime_checkable
+
+import structlog
+from ruamel.yaml import YAML
+
+from deepset_cloud_sdk._api.config import DEFAULT_WORKSPACE_NAME, CommonConfig
+from deepset_cloud_sdk._api.deepset_cloud_api import DeepsetCloudAPI
+from deepset_cloud_sdk.workflows.pipeline_client.models import (
+    IndexConfig,
+    PipelineConfig,
+)
+
+logger = structlog.get_logger(__name__)
+
+
+@runtime_checkable
+class PipelineProtocol(Protocol):
+    """Protocol defining the required methods for a pipeline object."""
+
+    def dumps(self) -> str:
+        """Convert the pipeline to a YAML string.
+
+        :return: YAML string representation of the pipeline
+        """
+        ...
+
+    def add_component(self, name: str, instance: Any) -> None:
+        """Add a component to the pipeline.
+
+        :param name: Name of the component
+        :param instance: Component instance to add
+        """
+        ...
+
+
+def _enable_publish_to_deepset() -> None:
+    """Add publish methods to the Haystack Pipeline and AsyncPipeline classes.
+
+    This function is called by deepset_sdk.init() to set up the SDK.
+    Users should not call this function directly.
+    """
+    try:
+        from haystack import AsyncPipeline as HaystackAsyncPipeline
+        from haystack import Pipeline as HaystackPipeline
+    except ImportError as err:
+        raise ImportError("Can't import Pipeline or AsyncPipeline, because haystack-ai is not installed.") from err
```

Comment:
```suggestion
        raise ImportError("Can't import Pipeline or AsyncPipeline, because haystack-ai is not installed. Run 'pip install haystack-ai'") from err
```